samtranslator/internal/deprecation_control.py

```diff
@@ -0,0 +1,44 @@
+"""
+Utils for deprecating our code using warning.warn().
+The warning message is written to stderr when shown.
+
+For the difference between DeprecationWarning
+and other deprecation warning classes, refer to
+https://peps.python.org/pep-0565/#additional-use-case-for-futurewarning
+
+If external packages import deprecated/pending-deprecation
+interfaces, it is their responsibility to detect and remove them.
+"""
+import warnings
+from functools import wraps
+from typing import Callable, Optional, TypeVar
+
+RT = TypeVar("RT")  # return type
+
+
+def _make_message(message: str, replacement: Optional[str]) -> str:
+    return f"{message}, please use {replacement}" if replacement else f"{message} and there is no replacement."
```

Comment:
I would get rid of the "and there is no replacement". It's normal to deprecate things and not expect it to be replaced. Also prevents confusion if someone just forgot to add the `replacement`.

Reply:
👍